staging/src/k8s.io/code-generator/pkg/util/build.go

```diff
@@ -57,7 +60,7 @@ func hasSubdir(root, dir string) (rel string, ok bool) {
 // BoilerplatePath uses the boilerplate in k8s.io/code-generator.
 // If it does not exist, it uses the boilerplate in k8s.io/kubernetes.
 func BoilerplatePath() string {
-	boilerPlatePath := "k8s.io/kubernetes/vendor/k8s.io/code-generator/hack/boilerplate.go.txt"
+	boilerPlatePath := reflect.TypeOf(Empty{}).PkgPath() + "/../../hack/boilerplate.go.txt"
```

Comment:
would `TypeOf(BoilerplatePath)` do as well?

Reply:
please use filepath.Join plus filepath.DirName

Reply:
or path.Join, I guess go package path do not use filepath?

Reply:
> would TypeOf(BoilerplatePath) do as well?

that would be a case of infinite recursion :)

Reply:
Doesn't work, unless I am missing something here.

https://play.golang.org/p/yuATNka3xYk

Reply:
a pity. then `empty` is fine.

Reply:
the type of Foo is an unnamed type. I guess that's the reason it does not work.

Compare https://play.golang.org/p/1toc2gKwkJM which works.

Reply:
But don't follow my example in that playground. Just for demonstration. It's ugly. Keep the `empty` ;-)